Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,243 @@
+    LogMsg "Current distro: $DISTRO"
+    for distro in "${supported_distro_list[@]}"
+    do
+        if [[ "$distro"* == $DISTRO ]]; then
+        LogMsg "Supported Distro family"
+        break
+    else
+        LogErr "Unsupported Distro family"
+        SetTestStateSkipped
+        exit 0
+    done
+}
+
+function verify_da_pid() {
+    [ ! -f $da_pid_file ] && LogErr "$da_pid_file does not exist." && SetTestStateFailed && exit 0
+
+    if x=$(sed -n '/^[1-9][0-9]*$/p;q' $da_pid_file); then
+        strings /proc/$x/cmdline | grep "microsoft-dependency-agent-manager" && LogMsg "PID matched" || (LogErr "PID not matched" && SetTestStateFailed && exit 0)
+    else
+        LogErr "PID not found in DA_PID"
+        SetTestStateFailed
+        exit 0
+    fi
+}
+
+function check_prereqs() {
+    LogMsg "Checking for preconditions"
+
+    if [[ $EUID -ne 0 ]]; then
+        LogErr "This script must be run as root" 
+        SetTestStateFailed
+        exit 0
+    fi
+
+    verify_directory
+    verify_distro
+
+    LogMsg "Preconditions met"
+}
+
+function download_da_linux_installer() {
+    LogMsg "Download Dependency Agent Linux installer"
+
+    curl -L https://aka.ms/dependencyagentlinux -o InstallDependencyAgent-Linux64.bin
+
+    chmod +x InstallDependencyAgent-Linux64.bin
+}
+
+function verify_install_da() {
+    ./InstallDependencyAgent-Linux64.bin -vme
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        LogErr "Install failed with exit code ${ret}"
+        SetTestStateFailed
+        exit 0
+    else
+        LogMsg "Installed with exit code 0"
+    fi
+
+    [ ! -f "/opt/microsoft/dependency-agent/uninstall" ] && LogErr "/opt/microsoft/dependency-agent/uninstall does not exist." && SetTestStateFailed && exit 0
+    [ ! -f "/etc/init.d/microsoft-dependency-agent" ] && LogErr "/etc/init.d/microsoft-dependency-agent does not exist." && SetTestStateFailed && exit 0
+    [ ! -f "/var/opt/microsoft/dependency-agent/log/install.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/install.log does not exist." && SetTestStateFailed && exit 0
+
+    bin_version=$(./InstallDependencyAgent-Linux64.bin --version | awk '{print $5}')
+    install_log_version=$(sed -n 's/^Dependency Agent version\.revision: //p' /var/opt/microsoft/dependency-agent/log/install.log)
+    if [ "$bin_version" -ne "$install_log_version" ]; then
+        LogErr "Version mismatch between bin version($bin_version) and install log version($install_log_version)"
+        SetTestStateFailed
+        exit 0
+    else
+        LogMsg "Version matches between bin version and install log version"
+    fi
+
+    LogMsg "Install tests passed successfully"
+}
+
+function verify_uninstall_da() {
+    /opt/microsoft/dependency-agent/uninstall
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        LogErr "Uninstall failed with exit code ${ret}"
+        SetTestStateFailed
+        exit 0
+    else
+        LogMsg "Uninstalled with exit code 0"
+    fi
+
+    verify_directory
```

Comment:
Recommendation: `verify_directory` could probably use a better name. Looking here in the code where it is called, the name `verify_directory` doesn't really tell me anything about **what** it does. From the name, it feels like there is a missing parameter, the name of the directory to be verified.

Picking "good" function names, ones that convey **what** the function does, while letting the human reader ignore the **how** it does it, is partially an art.

Reply:
Changed the function name to verify_directory_exists